network-peering/cmd/resource/resource.go

```diff
@@ -0,0 +1,196 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	matlasClient "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	defaultProviderName := "AWS"
+	projectID := *currentModel.ProjectId.Value()
+	peerRequest := matlasClient.Peer{
+		ContainerID: *currentModel.ContainerId.Value(),
+	}
+
+	region := currentModel.AccepterRegionName.Value()
+	if region == nil || *region == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("`accepter_region_name` must be set when `provider_name` is `AWS`")
```

Comment:
Same question as previously, does the provider have to be AWS if we're using the CloudFormation resource? Genuinely not sure.